Lib/fnmatch.py

```diff
@@ -77,23 +77,32 @@ def translate(pat):
     There is no way to quote meta-characters.
     """
 
-    STAR = object()
-    parts = _translate(pat, STAR, '.')
-    return _join_translated_parts(parts, STAR)
+    parts, indices = _translate(pat, '*', '.')
+    return _join_translated_parts(parts, indices)
 
+_re_setops_sub = re.compile(r'([&~|])').sub
+_re_escape = functools.lru_cache(maxsize=32768)(re.escape)
 
 def _translate(pat, STAR, QUESTION_MARK):
     res = []
     add = res.append
+    indices = []
+
     i, n = 0, len(pat)
     while i < n:
         c = pat[i]
         i = i+1
         if c == '*':
+            # store the position of the wildcard
+            indices.append(len(res))
```

Comment:
Also `chunks.append` below.

Reply:
I doubt it will change much and I wouldn't need to add an aditional line. I'll consider it as a cleanup step.

Reply:
Probably not worth it given this is the only access.

Reply:
I've confirmed with my benchmarks that caching the attribute doesn't help. It would have helped in older Python versions but for recent ones (in PGO builds), it doesn't help.

Reply:
If this is insignificant (unsurprisingly), just resolve this then.